nuxt-app/layouts/default.vue

```diff
@@ -0,0 +1,23 @@
+<template>
+    <div>
+        <h1>Nuxt - {{  componentName }}</h1>
+        <pie-link href="/">Home Page</pie-link>
+        <pie-divider></pie-divider>
+        <slot />
+    </div>
+</template>
+
+<script setup lang="ts">
+import { useRoute } from 'vue-router';
+import { computed } from 'vue';
+const route = useRoute();
+
+const componentName = computed(() => {
```

Comment:
(as mentioned offline) I would suggest leveraging page meta data for managing page names, titles and any other page-specific data we may want to use in the layouts. 

We can use `definePageMeta` in each page component to set the data and then access each meta property in this layout via `route.meta.{property}`.

[Reference](https://nuxt.com/docs/guide/directory-structure/pages#page-metadata)